src/main/webapp/app/exercises/quiz/shared/questions/multiple-choice-question/multiple-choice-question.component.ts

```diff
@@ -88,7 +88,12 @@ export class MultipleChoiceQuestionComponent {
             return;
         }
         if (this.isAnswerOptionSelected(answerOption)) {
-            this.selectedAnswerOptions = this.selectedAnswerOptions.filter((selectedAnswerOption) => selectedAnswerOption.id !== answerOption.id);
+            this.selectedAnswerOptions = this.selectedAnswerOptions.filter((selectedAnswerOption) => {
+                if (answerOption.id) {
+                    return selectedAnswerOption.id !== answerOption.id;
+                }
+                return selectedAnswerOption !== answerOption;
```

Comment:
I have 1 question, does it work the same if we just use the object comparison regardless of whether there is an id or not? If it works the same, I think we can simplify the condition.

Reply:
Hi, I thought about this and came to the conclusion that I only should change the behavior for answer options without an id. Although answer options should not change after a quiz has been released, I don't trust it enough to compare the whole answer options. If you still think that I should change it, let me know. Then I will do so :)